### PR TITLE
Update guidelines to include BEM

### PR DESCRIPTION
**Pull request by jadobson**

### CODE REVIEW

File: scss.md [hunk 1]
```diff
@@ -1,60 +1,108 @@
...
-Welcome to the coding standards for (S)CSS. Here are some guidelines for creating elegant stylesheets that work well with git diffs.
-
+Welcome to our coding standards for writing (S)CSS.
 
 ## Coding style
 
-* Use `//` For comments, as these won't appear in the generated CSS
 * Use spaces instead of tabs with a 2 space indent
-* Use multiple lines for selectors
+* Use multiple lines for multiple selectors
```
Comment (phillipvose) [inline scss.md]: Might be worth expanding this with an example, such as:

* Use multiple lines for multiple selectors, e.g.
```
.selector1,
.selector2 {
```

File: scss.md [hunk 2]
```diff
@@ -1,60 +1,108 @@
...
 * Put spaces before opening `{` to aid readability
 * Put a space after `:` to aid readability
 * Styles should be 1 per line to aid readability
-* Use pixels for font sizing, as this allows for greater control and stops you having to deal with inheriting `em` sizes
-* Use **hex** codes as default colours, **RGBA** if you need to alter the opacity
-* Hex code colours should always use the full 6 alphanumeric syntax, this makes for easier find and replace
+* Use hyphens for CSS names (e.g. `.package-block`, not `.packageBlock`)
+* Avoid over qualifying selectors (e.g. `p.qualified {…}` should just be `.qualified`)
+* Use single quotes when referencing strings
 * Whenever using colours, use the colours defined in `app/assets/stylesheets/application/settings.scss` so that if we need to change any, it will change across the whole site
```
Comment (phillipvose) [inline scss.md]: Could we expand this to explain that colours should be used in sets, so primary background with primary highlight, etc?  As this allows us to keep within the accessibility guidelines?

File: scss.md [hunk 3]
```diff
@@ -1,60 +1,108 @@
...
+* Comments should only when absolutely neccessary (the CSS should explain itself), for example, if a unicode character is used `\u201D` a comment would be neccsary to explain what it is
```
Comment (phillipvose) [inline scss.md]: Missing 'be used':
`* Comments should only be used when absolutely necessary (the CSS should explain itself), for example, if a unicode character is used `\u201D` a comment would be necessary to explain what it is`

File: scss.md [hunk 4]
```diff
@@ -1,60 +1,108 @@
...
+## SCSS Specifics
 
 Same rules as regular CSS, with a few extensions:
 
-* When using nested selectors, go a maximum of 3 levels deep, any more and you should consider restructuring your HTML or CSS Selectors
+* When using nested selectors, go a maximum of 3 levels deep, any more and you should consider restructuring your HTML or CSS selectors
 * If using variables, define them at the top of the file, or if used in more than one file, define them in `app/assets/stylesheets/application/settings.scss`
-* If you are using the same patterns in lots of places (e.g. a clearfix) then move them into `app/assets/stylesheets/application/helpers.scss`
-
+* If you are using the same patterns in lots of places (e.g. a clearfix) move it into the helpers folder as a new component `app/assets/stylesheets/application/helpers/`
```
Comment (phillipvose) [inline scss.md]: instead of saying 'new component' could we say 'as a new helper/mixin'?

File: scss.md [hunk 5]
```diff
@@ -1,60 +1,108 @@
...
 
 ## Everything should be a component
 
 When writing your styles, imagine every block on the page broken down into a component, this will make it easier to create reusable styles. See [SMACSS](http://smacss.com/)
 
+When creating a new component, its name should be kept short (ideally below 12 characters) but descriptive enough so that it's clear what the component is.
+
+* Good: `.deal-pod`
+* Bad: `.pod` (not specific enough)
+* Bad: `.flight-and-holiday-deal-pod` (too long)
```
Comment (phillipvose) [inline scss.md]: godcat

File: scss.md [hunk 6]
```diff
@@ -1,60 +1,108 @@
...
-      font-size: 15px;  
-      color: $royalBlue; // Colour would be defined in "settings.scss"  
+    // This is a good example
+    .selector-1,
+    .selector-2 {
+      display: block;
+      left: 0;
+      top: 0;
+      font-size: 15px;
+      color: $default-body-font; // Colour would be defined in "settings.scss"
```
Comment (phillipvose) [inline scss.md]: You asked me to remind you that you need to put a full path here...?

giphy 1

**Comment by JamesCotterill** [comment]
:+1: Looks good to me

**Comment by jameswalsh13** [comment]
:open_hands: Looks decent :open_hands:

**Pull request merged**